### DISCUSSION TITLE
Write accented characters while typing text.

### DISCUSSION THREAD

**Discussion by frnoal** [freeplane/freeplane#1762] [Help] [answered]
Hello everybody.

It's been a few days since I installed a very recent version of Freeplane via SNAP on my Linux computer. However, I cannot write accented characters while typing text. When I try to do so, the accented letter simply doesn't appear. Interestingly, I can insert (paste with CTRL+V) accented text normally.

The application configuration appears to be completely correct. UTF-8 characters in PT_br (Brazilian Portuguese).

On the other hand, I have an older version of Freeplane installed on the same machine via APT. In this version, all accented characters are typed normally.

Does anyone have any suggestions?
Thanks for any help.

**Comment by macmarrum** [comment, marked as answer]
Hi @frnoal 

I don't use SNAP on my Linux system so it's difficult for me to compare our experience. But I can type accented characters just fine, so it looks like a SNAP-specific issue.

If you'd like to try the latest Freeplane version, it can be easily run from a "Downloads" directory. Once you download [freeplane_bin-1.11.11.zip](https://sourceforge.net/projects/freeplane/files/freeplane%20stable/freeplane_bin-1.11.11.zip/download) and unzip it, you can start Freeplane via `freeplane.sh` (either double-click it, or open a terminal in `freeplane-1.11.11` and type `./freeplane.sh` followed by `Enter`).

**Reply by frnoal** [reply]
Excellent idea!
Thanks!

**Comment by leo-puh** [comment]
I can confirm the same issue with Flatpak: diacrytics symbols does not appear. I tried one googled workaround by passing XMODIFIERS="" variable, which apparently led to 8-bit ASCII diacrytics symbol appear instead of UTF-8.
Meanwhile, 'native' compiled build works well. So I guess there is something wrong with snap and flatpak builds or sandboxing breaking diacrytics input.

**Comment by AlexandreBonneau** [comment]
Could that issue be re-opened?
I'm not using snap nor flatpak, but I'm using freeplane from the official Deb package and I can say that most of the time (95%) you can enter accentuated characters using dead keys.

However, on 2 different freeplane versions and on 2 different computers (one Debian stable (`1.12.6~upstream-1`), one Debian testing (`1.12.10-pre06~upstream-1`), using Kde), a few problems can arise.
1/ Currently on Debian testing, even after a few reboots, Freeplane `1.12.10-pre06~upstream-1` does not accept any dead keys anymore, hence it's not possible to add any accentued characters.
2/ Sometimes, but I could not find any particular reason, freeplane stops accepting _any_ characters whatsoever. From that state, two things are possible ; either you save your work using the mouse then close and reopen freeplane, OR use `ibus restart`, but that leads to weird things elsewhere in the system, and you have to have ibus installed for starters.

So, to sum up, the problem exists and is not limited to snap and flatpak, but also affect the official upstream deb package.

**Reply by AlexandreBonneau** [reply]
Somehow I found a workaround following this [12 years old](https://unix.stackexchange.com/a/98024) thread on stackoverflow.
If I run `XMODIFIERS="" freeplane`, I can enter accentuated characters again!
I'm not sure why it worked a few months ago, and now needs that.
I also tried other java apps, and the accentuated characters are also impossible to enter, so I guess this is not related to freeplane, but to the changing Debian testing environment.

**Comment by JAPES1993** [comment]
Robust solution: like Linux is.
When I (not long ago) entered the Linux world I encountered the same problem. 
I m pleased to share the solution I found within Linux: 
install a keyboard definition with dead keys.
Probably, some programs/versions have solved this internally: but only for this environment. 
Perhaps this is not the brightest solution. 
But the most robust for me.
Linux: keyboard: lay-out: + eng Intl. with dead keys
